src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -655,38 +666,96 @@ void DlgPrefSound::refreshDevices() {
     emit refreshInputDevices(m_inputDevices);
 }
 
-/**
- * Called when any of the combo boxes in this dialog are changed. Enables the
- * apply button and marks that settings have been changed so that
- * DlgPrefSound::slotApply knows to apply them.
- */
+/// Called when any of the combo boxes in this dialog are changed. Enables the
+/// apply button and marks that settings have been changed so that
+/// DlgPrefSound::slotApply knows to apply them.
 void DlgPrefSound::settingChanged() {
     if (m_loading) {
         return; // doesn't count if we're just loading prefs
     }
     m_settingsModified = true;
 }
 
-void DlgPrefSound::deviceSettingChanged() {
+void DlgPrefSound::deviceChanged() {
     if (m_loading) {
         return;
     }
+
+    DlgPrefSoundItem* pItem = qobject_cast<DlgPrefSoundItem*>(sender());
+    if (!pItem) {
+        return;
+    }
+    QHash<DlgPrefSoundItem*, QPair<SoundDeviceId, int>>* channels;
+    if (pItem->isInput()) {
+        channels = &m_selectedInputChannelIndices;
+    } else {
+        channels = &m_selectedOutputChannelIndices;
+    }
+    auto id = pItem->getDeviceId();
+    qWarning() << "     dlg: dev Changed" << id.name;
```

Comment:
Shouldn't this be `qDebug()` instead? Samoe goes for all the other log messages below.

Reply:
whoopsy, re-added the debug commit while merging 2.4 into the local branch.
I'll remove it again.